elasticreindexer/process/reindexerMultiWrites.go

```diff
@@ -0,0 +1,112 @@
+package process
+
+import (
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-tools-go/elasticreindexer/config"
+)
+
+type interval struct {
+	start int64
+	stop  int64
+}
+
+type reindexerMultiWrite struct {
+	indicesNoTimestamp   []string
+	indicesWithTimestamp []string
+	numParallelWrite     int
+	blockChainStartTime  int64
+
+	reindexerClient *reindexer
+}
+
+func NewReindexerMultiWrite(cfg *config.GeneralConfig) (*reindexerMultiWrite, error) {
+	ri, err := CreateReindexer(cfg)
+	if err != nil {
+		return nil, err
+	}
+	return &reindexerMultiWrite{
+		reindexerClient:      ri,
+		indicesNoTimestamp:   cfg.Indexers.IndicesConfig.Indices,
+		indicesWithTimestamp: cfg.Indexers.IndicesConfig.WithTimestamp.IndicesWithTimestamp,
+		numParallelWrite:     cfg.Indexers.IndicesConfig.WithTimestamp.NumParallelWrites,
+		blockChainStartTime:  cfg.Indexers.IndicesConfig.WithTimestamp.BlockchainStartTime,
+	}, nil
+}
+
+func (rmw *reindexerMultiWrite) ProcessNoTimestamp(overwrite bool, skipMappings bool) error {
+	for _, index := range rmw.indicesNoTimestamp {
+		err := rmw.reindexerClient.Process(overwrite, skipMappings, index)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (rmw *reindexerMultiWrite) ProcessWithTimestamp(overwrite bool, skipMappings bool) error {
+	intervals, err := computeIntervals(rmw.blockChainStartTime, int64(rmw.numParallelWrite))
+	if err != nil {
+		return err
+	}
+
+	for _, index := range rmw.indicesWithTimestamp {
+		wg := &sync.WaitGroup{}
+		wg.Add(rmw.numParallelWrite)
+
+		log.Info("starting reindexing", "index", index)
+
+		count := uint64(0)
+
+		for _, interv := range intervals {
+			go func(startTime, stopTime int64) {
+				defer wg.Done()
+				errIndex := rmw.processIndexWithTimestamp(index, overwrite, skipMappings, startTime, stopTime, &count)
+				if errIndex != nil {
+					log.Warn("rmw.processIndexWithTimestamp", "index", index, "error", errIndex.Error())
+				}
+				time.Sleep(time.Second)
+			}(interv.start, interv.stop)
+		}
+
+		wg.Wait()
+	}
+
+	return nil
+}
+
+func (rmw *reindexerMultiWrite) processIndexWithTimestamp(index string, overwrite bool, skipMappings bool, start, stop int64, count *uint64) error {
+	return rmw.reindexerClient.processIndexWithTimestamp(index, overwrite, skipMappings, start, stop, count)
+}
+
+func computeIntervals(blockchainStartTime int64, numIntervals int64) ([]*interval, error) {
+	currentTimestampUnix := time.Now().Unix()
+	startTime := blockchainStartTime
+
+	if startTime > currentTimestampUnix {
+		return nil, errors.New("blockchain start time is greater than current timestamp")
+	}
+	if numIntervals < 2 {
+		return []*interval{{
+			start: blockchainStartTime,
+			stop:  currentTimestampUnix,
+		}}, nil
+	}
+
+	difference := currentTimestampUnix - startTime
+
+	step := difference / numIntervals
+
+	intervals := make([]*interval, 0)
```

Comment:
I don't think this algorithm works as expected.
I've written this test:
```GO
func TestTimestampDiv(t *testing.T) {
	start := 1653974028
	end := 1653981227

	numIntervals := 5

	difference := end - start
	step := difference / numIntervals

	fmt.Printf("the step is %d\n", step)

	for i:=0;i<numIntervals;i++{
		fmt.Printf("step %d: from %d to %d\n", i, start, start + step)
		start += step
	}
}
```

output: 
```
the step is 1439
step 0: from 1653974028 to 1653975467
step 1: from 1653975467 to 1653976906
step 2: from 1653976906 to 1653978345
step 3: from 1653978345 to 1653979784
step 4: from 1653979784 to 1653981223
```

2 things to mention:
- because the result of the division is an integer number, there will be a reminder that is added to the last step. notice that `1653981223` is lower than `1653981227`. It's not a big issue, but it's good to consider
- step 0 for example ends at `1653975467` but the step 1 begins at `1653975467` which is the same timestamp and I think this will result in the same txs getting indexed both times

Reply:
Fixed and refactor
Also added an unit test